common/tor.c

```diff
@@ -230,8 +199,10 @@ static struct io_plan *tor_conn_init(struct io_conn *conn,
 	static struct tor_service_reaching reach;
 	static struct addrinfo *ai_tor;
 	reach.ld = ld;
-	getaddrinfo(tal_strdup(NULL, "127.0.0.1"), tal_strdup(NULL, "9051"),
-		    NULL, &ai_tor);
+
+	getaddrinfo(fmt_wireaddr_without_port(ld, ld->tor_serviceaddrs),
```

Comment:
If the strings being generated here are temporary and only used for this call, it is wasteful to keep them around while `ld` is alive, especially, `ld` is alive as long as the lightningd is running. So, best to make a `tal_tmpctx` within this function, free it at all `return` points, and use the `tmpctx` for temporary strings.